// components/createArticle/index.js
import React, { useEffect, useState , useContext} from 'react';
import styles from './index.css'
import {Link, Redirect} from "react-router-dom";
import ArticleForm from 'components/articleForm'
import useFetch from 'hooks/useFetch'
import {CurrentUserContext} from 'context/currentUser'
const Index = () => {
    const apiUrl = '/articles'
    const [{response, error}, doFetch] = useFetch(apiUrl)
    const errors = {}
    const [currentUserState] = useContext(CurrentUserContext)
    const [isSuccessfullSubmit, setIsSuccessfullSubmit] = useState(false)
    const initialValues = {
        title : '',
        description : '',
        body : '',
        tagList : []
    }

    const handleSubmit = article => {
       doFetch({
           method: 'post',
           data : {
               article
           }
       })
    }


    useEffect(() => {
        if(!response) {
            return
        }
        setIsSuccessfullSubmit(true)

    }, [response])
    if (currentUserState.isLoggedIn === false) {
        return <Redirect to="/"/>
    }
    if (isSuccessfullSubmit) {
        return <Redirect to= {`/article/${response.article.slug}`}/>
    }
    return (
        <div>
            <ArticleForm errors={(error && error.errors) || {}} initialValues={initialValues} onSubmit={handleSubmit}/> 
        </div>
       
       
    );

};

export default Index;
